jekyll/_cci2/docker-hub-pull-through-mirror.adoc

```diff
@@ -20,60 +20,136 @@ Starting https://www.docker.com/blog/scaling-docker-to-serve-millions-more-devel
+TIP: Setting up a paid account can be a good option to ease the rate limiting further. See also: https://www.docker.com/pricing.
 
-2. *Set up an independent Linux server* that has Docker installed.
+. *Set up an independent Linux server* that has Docker installed.
 +
-We set up this registry as an independent server (i.e. outside of CircleCI boxes) to avoid load on this cache server affecting other CircleCI Server services.
+We set up this registry as an independent server (i.e. outside of CircleCI boxes) to avoid load on this cache registry affecting other CircleCI Server services.
 +
-Assume that the IP address for this server is `192.0.2.1`.
+Assuming that the IP address for this server is `192.0.2.1`, the URL for the registry to set up is `http://192.0.2.1`. The URL will be needed later to arm Nomad clients and the VM Service.
 
-3. Run the command below to *start the cache server*.
+. Run the command below to *start the registry*.
 +
 NOTE: Replace `DOCKER_HUB_USERNAME` and `DOCKER_HUB_ACCESS_TOKEN` with the username for the Docker Hub account and the access token you obtained through https://hub.docker.com/settings/security, respectively.
 +
-`sudo docker run -d -p 80:5000 --restart=always --name=through-cache -e REGISTRY_PROXY_REMOTEURL="https://registry-1.docker.io" -e REGISTRY_PROXY_USERNAME=DOCKER_HUB_USERNAME -e REGISTRY_PROXY_PASSWORD=DOCKER_HUB_ACCESS_TOKEN registry`
+[source,bash]
+----
+sudo docker run \
+  -d \
+  -p 80:5000 \
+  --restart=always \
+  --name=through-cache \
+  -e REGISTRY_PROXY_REMOTEURL="https://registry-1.docker.io" \
+  -e REGISTRY_PROXY_USERNAME=DOCKER_HUB_USERNAME \
+  -e REGISTRY_PROXY_PASSWORD=DOCKER_HUB_ACCESS_TOKEN \
+  registry
+----
 
-4. Finally, make sure that the TCP port 80 (i.e., HTTP) is open and accessible. For better security, we recommend you open the port only to Nomad clients.
+. Finally, make sure that the TCP port 80 (i.e. HTTP) is open and accessible. For better security, we recommend that you only open the port to Nomad clients and VMs for `machine` executors and remote docker engines.
 +
 On AWS, for example, you will need to make sure that both Security Groups and the firewall at the OS level are configured correctly to accept connections from Nomad clients over HTTP.
 
-== Configure Nomad clients to use the pull through cache server (run this for _each_ Nomad client)
+=== Set up a secure pull through cache registry (HTTPS proxy for Docker Hub)
+
+Under certain circumstances (especially when you are enabling https://docs.docker.com/develop/develop-images/build_enhancements/[BuildKit] on `machine` executors and remote Docker engines), your pull through cache registry needs to accept connections over HTTPS. These instructions cover setting up a pull through cache that listens for HTTPS connections.
+
+. *Set up a Docker Hub account*, which will be used by the pull through cache registry to access Docker Hub.
```

Comment:
Can we call out here as a warning that any account they auth with will be accessible without explicit auth within their Docker jobs? This could be a real security concern for some folks.

Reply:
Good call. Will work on Monday.

 It should relate to [this comment](https://github.com/circleci/circleci-docs/pull/4906#discussion_r518734407), and it sounds better to rephrase this part and then relocate it to somewhere eye-catching.

Reply:
Done.

@rosieyohannan @ryanwohara May I ask your double check on the portion I've edited? I've added another tens of lines of sentences, and it should be better to peer reviewed. 🙏